File: toh.py
def towerhanoi(n:int,  src:str, dest:str, temp:str):
    if n == 1:
        print(f"Move disk 1 from {src} to {dest}")
    else:
        towerhanoi(n-1, src, temp, dest)
        print(f"Move disk {n} from {src} to {dest}")
        towerhanoi(n-1, temp, dest, src)
n=int(input("Enter the number of disks: "))
towerhanoi(n, 'A', 'C', 'B') # A, B and C are the names of the rods(src, dest and temp resp)

File: toh.c
#include <stdio.h>
void hanoi(int n, char src, char dst, char temp) {
   if(n == 1){
      printf("Move disk 1 from %c to %c\n", src, dst);
   }
   else{
      hanoi(n-1, src, temp, dst);
      printf("Move disk %d from %c to %c\n", n, src, dst);
      hanoi(n-1, temp, dst, src);
   }
}
int main() {
   int n;
   printf("Enter the no of the disc:");
 scanf("%d",&n);
 printf("\n");
   char src = 'A';
   char dst = 'B';
   char temp = 'C';
   //calling hanoi() method
   hanoi(n, src ,temp, dst);
}
